framework/src/modules/nft/method.ts

```diff
@@ -61,4 +70,81 @@ export class NFTMethod extends BaseMethod {
 
 		return userData.lockingModule;
 	}
+
+	public async destroy(
+		methodContext: MethodContext,
+		address: Buffer,
+		nftID: Buffer,
+	): Promise<void> {
+		const nftStore = this.stores.get(NFTStore);
+
+		const nftExists = await nftStore.has(methodContext, nftID);
+
+		if (!nftExists) {
+			this.events.get(DestroyEvent).log(
+				methodContext,
+				{
+					address,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_DOES_NOT_EXIST,
+			);
+
+			throw new Error('NFT substore entry does not exist');
+		}
+
+		const owner = await this.getNFTOwner(methodContext, nftID);
+
+		if (!owner.equals(address)) {
+			this.events.get(DestroyEvent).log(
+				methodContext,
+				{
+					address,
+					nftID,
+				},
+				NftEventResult.RESULT_INITIATED_BY_NONOWNER,
+			);
+
+			throw new Error('Not initiated by the NFT owner');
+		}
+
+		const userStore = this.stores.get(UserStore);
+		const userKey = userStore.getKey(owner, nftID);
```

Comment:
The goal was to encapsulate key concatenation logic, but I'm afraid I was not able to understand the point about making it more generic, could you please explain it more.

Reply:
I was just noticing, by looking at the way the function is defined, it looks that it will serve only for the purpose of concatenating `receivingChainID` and `nftID`, while you use it also in other cases (for example, to concatenate the address of the `owner` and the `nftID`). But it serves it's purpose of concatenation, as you say, so it's fine.